src/components/ui4/ButtonsViewUi4.tsx

```diff
@@ -52,57 +56,106 @@ interface Props {
 export const ButtonsViewUi4 = React.memo(
   ({
     absolute = false,
-    fade,
     primary,
     secondary,
     secondary2,
     tertiary,
     layout = 'column',
-    parentType
+    parentType,
+    animDistanceStart
   }: Props) => {
-    const numButtons = [primary, secondary, secondary2, tertiary].filter(
+    const buttonInfos = [primary, secondary, secondary2, tertiary].filter(
       key => key != null
-    ).length
-    if (numButtons === 1) layout = 'solo'
+    )
+    if (buttonInfos.length === 1) layout = 'solo'
 
     const spacing = <Space around={INTER_BUTTON_SPACING_REM / 2} />
 
-    const renderButton = (type: ButtonTypeUi4, buttonProps?: ButtonInfo) => {
+    const renderButton = (
+      type: ButtonTypeUi4,
+      buttonProps?: ButtonInfo,
+      index: number = 0
+    ) => {
       if (buttonProps == null) return null
       const { label, onPress, disabled, spinner, testID } = buttonProps
 
+      const distance =
+        animDistanceStart != null
+          ? animDistanceStart + index * ANIM_DISTANCE_INCREMENT
+          : undefined
+      // TODO: Sync EdgeAnim w/ GUI
+      // const disableAnimation = Platform.OS === 'android'
+
       return (
-        <ButtonUi4
-          layout={layout}
-          label={label}
-          onPress={onPress}
-          type={type}
-          disabled={disabled}
-          spinner={spinner}
-          testID={testID}
-        />
+        <MaybeEdgeAnim
+          when={animDistanceStart != null}
+          // disableAnimation={disableAnimation} // TODO
+          enter={{ type: 'fadeInDown', duration: ANIM_DURATION, distance }}
+        >
+          <ButtonUi4
+            layout={layout}
+            label={label}
+            onPress={onPress}
+            type={type}
+            disabled={disabled}
+            spinner={spinner}
+            testID={testID}
+          />
+        </MaybeEdgeAnim>
       )
     }
+
+    const hasPrimary = primary != null
+    const hasSecondary = secondary != null
+    const hasSecondary2 = secondary2 != null
+    const hasTertiary = tertiary != null
+
     return (
       <StyledButtonContainer
         absolute={absolute}
         layout={layout}
         parentType={parentType}
       >
-        {renderButton('primary', primary)}
-        {primary != null && secondary != null ? spacing : null}
-        {renderButton('secondary', secondary2)}
-        {secondary != null && secondary2 != null ? spacing : null}
-        {renderButton('secondary', secondary)}
-        {tertiary != null ? spacing : null}
-        {renderButton('tertiary', tertiary)}
+        {hasPrimary && (
+          <>
+            {renderButton('primary', primary, 0)}
+            {(hasSecondary || hasSecondary2 || hasTertiary) && spacing}
```

Comment:
While I won't ask for changes just yet, this is really starting to get messy and very bug prone as I predicted. I'd sooner have multiple ButtonsView containers for a few common layouts.

Reply:
I agree, a new layer in the hierarchy is worthwhile soon